osd: minor performance improvements and fixes

xiexingguo opened the pull request.

xiexingguo commented:
@athanatos Please take a look. Thanks!

athanatos commented on src/osd/OSD.h:
```diff
@@ -949,13 +949,13 @@ class OSDService {
     _maybe_queue_recovery();
   }
-  void clear_queued_recovery(PG *pg, bool front = false) {
+  void clear_queued_recovery(PG *pg) {
```
Actually, this one may come to life again soon, please remove this commit.

athanatos commented on src/common/hobject.h:
```diff
@@ -250,7 +250,8 @@ struct hobject_t {
   }
   void set_bitwise_key_u32(uint32_t value) {
     hash = _reverse_bits(value);
-    build_hash_cache();
```
Please add a comment to this and build_hash_cache() to ensure that we remember to update both.

xiexingguo commented:
@athanatos Done and thanks for the confirmation.

athanatos commented:
@xiexingguo I haven't reviewed those 4 new ones.  I'd appreciate it if you did not add more commits to a PR after I've reviewed the PR and added the needs-qa label...  Please remove them and ping me so that I can re-add the needs-qa label.

xiexingguo commented:
> I'd appreciate it if you did not add more commits to a PR after I've reviewed the PR and added the needs-qa label...

@athanatos Sure.
Moved those changes into https://github.com/ceph/ceph/pull/10548, and please take a look at them too. Thanks!

yuriw commented:
http://pulpito.ceph.com/yuriw-2016-08-05_14:30:29-rados-wip-yuri-testing_2016_08_05-distro-basic-smithi/
